nomad/stream/event_publisher.go

```diff
@@ -0,0 +1,220 @@
+package stream
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/nomad/nomad/structs"
+
+	"github.com/hashicorp/go-hclog"
+)
+
+const (
+	DefaultTTL = 1 * time.Hour
+)
+
+type EventPublisherCfg struct {
+	EventBufferSize int64
+	EventBufferTTL  time.Duration
+	Logger          hclog.Logger
+	OnEvict         EvictCallbackFn
+}
+
+type EventPublisher struct {
```

Comment:
nitpicky semantics - this seems more like a EventBroker than a publisher - as it manages subscriptions too.

Reply:
EventBroker now! :+1: